sympy/functions/combinatorial/factorials.py

```diff
@@ -1057,3 +1058,103 @@ def _eval_is_nonnegative(self):
                 return True
             elif k.is_even is False:
                 return  False
+
+class multinomial(CombinatorialFunction):
+    r"""Implementation of the multinomial coefficient. They can be expressed as
+    a product of binomial coefficients or of factorials:
+
+    .. math:: \frac{(k_1+k_2+\cdots+k_m)!}{k_1!\, k_2! \cdots k_m!} = {k_1\choose k_1}{k_1+k_2\choose k_2}\cdots{k_1+k_2+\cdots+k_m\choose k_m}
+
+    The multinomial coefficients have a direct combinatorial interpretation,
+    as the number of ways of depositing n distinct objects into m distinct bins,
+    with k1 objects in the first bin, k2 objects in the second bin, and so on.
+
+    For the sake of convenience for any negative integer `k` this function
+    will return zero.
+
+    To expand the multinomial when `k` contains a symbol, use either
+    ``expand_func()`` or ``expand(func=True)``. The former will keep
+    the polynomial in factored form while the latter will expand the
+    polynomial itself. See examples for details.
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, Rational, multinomial, expand_func, symbols, gamma
+    >>> n = Symbol('n', integer=True, positive=True)
+
+    >>> multinomial(15, 8)
+    490314
+
+    Comparing Coefficents generated with the multinomial function:
+
+    >>> a, b, c = symbols('a b c')
+    >>> ((a+b+c)**3).expand(func=True)
+    a**3 + 3*a**2*b + 3*a**2*c + 3*a*b**2 + 6*a*b*c + 3*a*c**2 + b**3 + 3*b**2*c + 3*b*c**2 + c**3
+
+    The Coefficient when of 2 1 0 corresponding to the second term is 3
+
+    >>> multinomial(2, 1, 0)
+    3
+
```

Comment:
Here is where a comment about the generalization of the multinomial should be made since you are no longer dealing with "coefficients of a polynomial" (though I am not versed in how to interpret the result of non-integral args).

Reply:
I am not particularly qualified to interpret that either 😅  ... I'll look around established resources and find something that fits 👍

Reply:
non integrals can still be used to represent powers in expansion series.
like[ (1 + x)<sup>2/3</sup> = x<sup>2/3</sup> + (2/3) x<sup>1</sup> + 1/3 x<sup>0</sup>](http://www.wolframalpha.com/input/?i=(1%20%2Bx%20)%5E%202%2F3)
so to find out the coefficient of the second term x<sup>1</sup>, we write a multinomial (1 , a )   
such that 1 + a = n (the power of the expansion series )
so 2/3 -1 = a = -1/3 hence ... 
[ multinomial(1, -1/3) = 2/3 ](http://www.wolframalpha.com/input/?i=multinomial(1%2C%20-1%2F3)) which is the coefficient of the second term ...

is this satisfactory ... if so ... I'll phrase it in simple terms and include it in the docs.

Reply:
Thanks, that helps. (But someone should probably be using `binomial` for this, right?)

Reply:
I would've given a more general example but it comes out really complicated 😅 . I mean with the expansions and everything. 😄

Reply:
@smichr here's what i've drafted. Tell me what you think, I'll commit after that.
```python3
"""
A series `(1 + x)^n`, where `n` is 2/3 when expanded, would yield
`x^(2/3) + (2/3) x^1 + (1/3) x^0`. To find the coeffient of `x^1`, one needs to use
`multinomial(1, a)` where `1 + a = n; a = -1/3`
"""
```
and added this before the examples
```python3
"""
To interpret using non-integral values of `k` one can consider an expansion series of power `n`,
`(a_1 + a_2 + ... + a_m) ^ n` such that `(k_1 + k_2 + ... + k_m) = n` where `k_i` is the
coefficient of `a_i` after expansion.
"""
```

Reply:
This seems to be a very obtuse -- though mathematically correct -- way to get the coefficient of a term in a univariate expansion. It looks like you using the binomial answer to instruct someone how to get that answer from multinomial. What if we had `(x + y)**(2/3)`? What would `multinomial(1/3,1/3)` correspond to? I am not sure that `series` works with multivariable expressions, but we could replace `y` with `2` and see the answer -- whatever it is -- to the last question.

Reply:
Technically ...
![Screen Shot 2020-03-30 at 2 25 06 PM](https://user-images.githubusercontent.com/28682735/77894030-5714e400-7292-11ea-976c-5cdcc384df4f.png) [check here](http://www.wolframalpha.com/input/?i=(x%20%2B%20y)**(2%2F3))
so multinomial(1/3 , 1/3) would give you the coefficient of 
x<sup>1/3</sup> y<sup>1/3</sup> <sup>2/3</sup><b>C</b><sub>1/3</sub> am I right ?
i.e [<sup>2/3</sup><b>C</b><sub>1/3</sub> == 1.3209](http://www.wolframalpha.com/input/?i=binomial(2%2F3%20%2C%201%2F3))
which is also equal to [multinomial(1/3, 1/3) == 1.3209](http://www.wolframalpha.com/input/?i=multinomial(1%2F3%2C%201%2F3))
This should also work when generalised to more than 2 variables, while the interpretation gets more convoluted the method is still the same.
if this seems obtuse, what other examples do you suggest I incorporate ? @smichr